[BUG] OpenImageIO/strutil.h macro pollution

**Describe the bug**

`OpenImageIO/strutil.h` includes `OpenImageIO/detail/farmhash.h`, which defines macros for many common names, including [`fmix`](https://github.com/OpenImageIO/oiio/blob/master/src/include/OpenImageIO/detail/farmhash.h#L1142). This pollutes the namespace, breaking client code (in this case, Gaffer).

**To Reproduce**

Compiling this code :

```
#include "OpenImageIO/strutil.h"

namespace Test
{
    uint64_t fmix( uint64_t k );
}
```

Results in an error along these lines :

```
test.cpp:5:10: error: ‘farmhash’ has not been declared
    5 | uint64_t fmix( uint64_t k );
```

**Expected behavior**

Compilation succeeds. OIIO's public headers only define macros with an OIIO prefix.

**Evidence**

https://github.com/OpenImageIO/oiio/blob/master/src/include/OpenImageIO/detail/farmhash.h#L1142

**Platform information:**
 - OIIO branch/version: Version 2.2.15.1.
 - OS: CentOS 7.
 - C++ compiler: GCC 9.3.1
 - Any non-default build flags when you build OIIO:

--- a/src/include/OpenImageIO/detail/farmhash.h
+++ b/src/include/OpenImageIO/detail/farmhash.h
@@ -2096,18 +2096,38 @@ STATIC_INLINE uint128_t Fingerprint128(const char* s, size_t len) {
 // }  // namespace NAMESPACE_FOR_HASH_FUNCTIONS
 } /*end namespace farmhash*/ 
 
+// Undefine any of the poorly namespaced things
 #undef Fetch
 #undef Rotate
 #undef Bswap
+#undef fmix
 #undef DebugTweak
 #undef uint128_t
 #undef Uint128
+#undef CopyUint128
 #undef Uint128Low64
 #undef Uint128High64
 #undef Hash128to64
 #undef Hash64WithSeeds
 #undef x86
 #undef x86_64
+#undef is_64bit
+#undef can_use_ssse3
+#undef can_use_sse41
+#undef can_use_sse42
+#undef can_use_aesni
+#undef can_use_avx
+#undef bswap_32
+#undef bswap_64
+#undef STATIC_INLINE
+#undef uint32_in_expected_order
+#undef uint64_in_expected_order
+#undef debug_mode
+#undef PERMUTE3
+#undef Mulc1
+#undef Mulc2
+#undef Murk
+#undef Chunk
 
 OIIO_NAMESPACE_END
 

